src/main/scala/scorex/core/network/NodeViewSynchronizer.scala

```diff
@@ -269,7 +288,9 @@ class NodeViewSynchronizer[TX <: Transaction,
           } else {
             log.info(s"Peer $peer has not delivered asked modifier ${encoder.encode(modifierId)} on time")
             penalizeNonDeliveringPeer(peer)
-            deliveryTracker.reexpect(Some(peer), modifierTypeId, modifierId)
+            if (deliveryTracker.reexpect(Some(peer), modifierTypeId, modifierId).isFailure) {
+              statusKeeper.remove(modifierId)
```

Comment:
So status clearing works only if we asked a concrete peer, thus if random peers did not deliver we do not clear the status. Looks buggy, and I would like to propose a serious refactoring of this method and corresponding functionality.